tests/integration/delete_user.t

```diff
@@ -0,0 +1,123 @@
+	delete => 'on',
+	action => 'process',
+	type => 'edit',
+	userid => 'tests'
+);
+
+#checking if the delete button exist
+like($response_edit->content, qr/\Q$words[0]\E/i, "the delete button does exist");
+
+#deleting the account
+my $before_delete_ts = time();
+my $response_delete = $ua->post($url_delete, \%delete_form);
+#checking if we are redirected to the account deleted page
+like($response_delete->content, qr/\Q$words[1]\E/i, "the account was deleted");
+
+#waiting the deletion task to be done
+my $jobs = $minion->jobs({tasks => ["delete_user_task"]});
+#iterate on job
+while (my $job = $jobs->next) {
+	#only those who were created after the timestamp
+	my $waited = 0;    #waiting time
+	if ($job->created > $before_delete_ts) {
+		#waiting the job to be done
+		while ($job->state == "inactive" or $job->state == "active" or $waited < 200) {
+			sleep(2);
+			$waited++;
+		}
+	}
+}
+#checking if there is still delete_user_task jobs
+$jobs = $minion->jobs({tasks => ["delete_user_task"]});
+my $jobs_count = 0;
+while (my $job = $jobs->next) {
+	$jobs_count++;
+}
+ok($jobs_count == 0, "delete user task is finished");
+
+#user sign out of its account
+my %signout_form = (
+	length => "logout",
+	".submit" => "Sign out"
+);
+my $url_signout = construct_test_url("/cgi/session.pl", "world");
+my $response_signout = $ua->post($url_signout, \%signout_form);
+
+like($response_signout->content, qr/\Q$words[2]\E/i, "the user signed out");
+
+#admin ua checking if the account is well deleted
+my $response_userid = $admin->get($url_userid);
+
+my $url_email = construct_test_url('/cgi/user.pl?type=edit&userid=bob@test.com', "world");
+my $response_email = $admin->get($url_email);
+
+my $url_contributor = construct_test_url("/contributor/tests", "world");
+my $response_contributor = $admin->get($url_contributor);
+
+#checking if the edit page of the common ua is well deleted
+like($response_userid->content, qr/\Q$words[3]\E/i, "the userid edit page is well deleted");
+#checking if the edit page of the common ua is well deleted
+like($response_email->content, qr/\Q$words[3]\E/i, "the email edit page is well deleted");
+#checking if the edit page of the common ua is well deleted
+like($response_contributor->content, qr/\Q$words[4]\E/i, "the contributor page of the ua is well deleted");
+
+#checking if an ua can reconnect with the deleted account ids
+my $url_login = construct_test_url("/cgi/login.pl", "world");
+my %login_form = (
+	user_id => "tests",
+	password => "testtest",
+	submit => "Sign in"
+);
+my $response_login = $ua->post($url_login, \%login_form);
+like($response_login->content, qr/\Q$words[5]\E/i, "an user can't login with the deleted account ids");
+
+#checking if the added product has been anonymized
+my $url_product = construct_test_url("/cgi/product.pl?type=edit&code=2000000000001", "world");
+my $response_product = $admin->get($url_product);
+like($response_product->content, qr/\Q$words[6]\E/i, "the product has been anonymized");
```

Comment:
You would rather go to "/api/v2/product/2000000000001", decode the json response you get and check the creator and editors_tags fields.

Reply:
Is it a better way ? I tried it and it works well